sorc/ncep_post.fd/MDL2P.f

```diff
@@ -4318,6 +4336,93 @@ SUBROUTINE MDL2P(iostatusD3D)
             enddo
          endif
       ENDIF
+
+!
+!------------------------------------------------------------------------------------
+!*** K. ASMAR - COMPUTE VELOCITY POTENTIAL AND STREAMFUNCTION FROM MB WINDS USL/VSL
+!*** SPECTRALLY TRUNCATE WIND VECTOR FIELDS ON GLOBAL CYLINDRICAL GRID
+!*** TO COMPUTE VELOCITY POTENTIAL AND STREAMFUNCTION.
+!------------------------------------------------------------------------------------
+!
+	IF (IGET(1021) > 0 .OR. IGET(1022) > 0) THEN												
+          IF(IDRT == 0)THEN
+	    JCAP = (JM-3)/2
+	  ELSE
+	    JCAP = JM-1
+	  ENDIF
+          CALL SPTRUNV(0,JCAP,IDRT,IM,JM,								&
```

Comment:
@WenMeng-NOAA Should I include in the new subroutine the collection of the USL/VSL winds in the full domain (as done [here](https://github.com/KarinaAsmar-NOAA/UPP/blob/9246b809fb2f0961b533184f6f0ddc36ac4925e5/sorc/ncep_post.fd/MDL2P.f#L1781)? Or should that be done in MDL2P.f or in a separate subroutine?

Reply:
@KarinaAsmar-NOAA 
@WenMeng-NOAA 
My understanding, Wen please verify, after we obtained USL/VSL in MDL2P, call a new subroutine outside MDL2P and move the code of calculating velocity potential and stream function to that new subroutine, which includes
1. call COLLECT_ALL
2. call SPTRUNV with if (me .eq. 0) only
3. return the full domain CHI and PSI back to MDL2P 
Back in MDL2P assign the subset of CHI and PSI to GRID1

Reply:
@JesseMeng-NOAA Could the USL/VSL collection be moved into new subroutine? If not, it can remain in MDL2P.f. Your proposed new subroutine look good for me.

Reply:
@JesseMeng-NOAA Where should the USL/VSL winds be saved to all LSM levels (e.g, in MDL2P.f or the new subroutine)? At the moment, they are computed inside the LP=1,LSM loop in MDL2P.f.

Reply:
@KarinaAsmar-NOAA @WenMeng-NOAA 
Calculate and save USL/VSL in MDL2P.  Pass (USL, VSL, CHI, PSI) to the new subroutine along with anything else SPTRUNV needs. COLLECT_ALL can be done in the new subroutine.

Reply:
@WenMeng-NOAA @JesseMeng-NOAA The latest changes have been pushed to the branch. The new subroutine CALCHIPSI.f collects the winds and calls sptrunv. VPOT and STRM are then sliced and saved in MDL2P.f. The plots comparing the new changes with CFS are here: 
[VPOT and STRM Plots v2.pptx](https://github.com/user-attachments/files/16414943/VPOT.and.STRM.Plots.v2.pptx)